### PR TITLE
Fix bug in AMFClassifier and finish AMFRegressor (#1166)

### PR DESCRIPTION
**Pull request by smastelini**
Thanks to a comment in the Fellowship of Online Machine's discord, I was able to track a bug affecting both the classifier and regression versions of AMF.

In my preliminary tests, the predictive performance improved in both cases. @kenzabenjelloun, thanks for your work on the AMFRegressor. I think the bug you mentioned in the previous PR might have been fixed :D

### CODE REVIEW

File: docs/releases/unreleased.md [hunk 1]
```diff
@@ -11,3 +11,9 @@ Calling `learn_one` in a pipeline will now update each part of the pipeline in t
 ## forest
 
 - Fixed issue with `forest.ARFClassifier` which couldn't be passed a `CrossEntropy` metric.
+- Fixed bug in `forest.AMFClassifier`.
```
Comment (MaxHalford) [inline docs/releases/unreleased.md:14]:
```suggestion
- Fixed a bug in `forest.AMFClassifier` which slightly improves predictive accurary.
```

File: river/forest/aggregated_mondrian_forest.py [hunk 2]
```diff
@@ -217,3 +218,116 @@ def predict_proba_one(self, x):
...
+    """Aggregated Mondrian Forest regressor for online learning.
+
+    This algorithm is truly online, in the sense that a single pass is performed, and that
+    predictions can be produced anytime.
+
+    Each node in a tree predicts according to the average of the labels it contains.
+    The prediction for a sample is computed as the aggregated predictions of all the subtrees
+    along the path leading to the leaf node containing the sample. The aggregation weights are
+    exponential weights with learning rate ``step`` using a squared loss when ``use_aggregation``
+    is ``True``.
```
Comment (MaxHalford) [inline river/forest/aggregated_mondrian_forest.py:233]:
```suggestion
    exponential weights with learning rate `step` using a squared loss when `use_aggregation`
    is `True`.
```

File: river/tree/mondrian/mondrian_tree_nodes.py [hunk 3]
```diff
@@ -366,3 +366,150 @@ class MondrianBranchClassifier(MondrianNodeClassifier, MondrianBranch):
...
+
+
+class MondrianNodeRegressor(MondrianNode):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.n_samples = 0
+        self.mean = stats.Mean()
+
+    def replant(self, leaf: MondrianNodeRegressor, copy_all: bool = False):
```
Comment (MaxHalford) [inline river/tree/mondrian/mondrian_tree_nodes.py:378]: Love this method name

File: river/tree/mondrian/mondrian_tree_nodes.py [hunk 4]
```diff
@@ -366,3 +366,150 @@ class MondrianBranchClassifier(MondrianNodeClassifier, MondrianBranch):
...
+
+        Parameters
+        ----------
+        sample_value
+            Label of a given sample.
+        use_aggregation
+            Whether to use aggregation or not during computation (given by the tree).
+        step
+            Step parameter of the tree.
+        """
```
Comment (MaxHalford) [inline river/tree/mondrian/mondrian_tree_nodes.py:421]:
```suggestion

        """
```

Normally an empty should be placed at the end of each docstring

File: river/forest/aggregated_mondrian_forest.py (hunk 2 above)
Comment (smastelini) [reply river/forest/aggregated_mondrian_forest.py:233]: I didn't see that! Thanks!

File: river/tree/mondrian/mondrian_tree_nodes.py (hunk 4 above)
Comment (smastelini) [reply river/tree/mondrian/mondrian_tree_nodes.py:421]: I will check that as soon as I get to my computer

**Pull request merged**